lhotse/recipes/aishell2.py

```diff
@@ -15,6 +15,59 @@
 from lhotse.utils import Pathlike
 
 
+def text_normalize(line: str):
+    """
+    Modified from https://github.com/wenet-e2e/wenet/blob/main/examples/multi_cn/s0/local/aishell2_data_prep.sh#L50
+
+    """
+    line = line.replace("Ａ", "A")
+    line = line.replace("Ｔ", "T")
+    line = line.replace("Ｍ", "M")
+    line = line.replace("𫖯", "頫")
+    """
+    The below code is to only remove "'" of mandarin.
+    The "'" of english remains unchanged.
+
+    for example:
+    IC0010W0228	Here's
+    IC0012W0161	I'm
+    IC0013W0018	It's
+    IC0017W0126	Nothing'sGChange
+    IC0020W0392	She's
+    IC0022W0444	That's
+    IC0073W0058	搬不走的要及时'关停并转'
+    IC0085W0187	帮我放一首歌Let's
+    IC0392W0410	对低收入群体的帮助也更大'
+    IC0975W0451	明年二月底'小成'
+    ID0114W0368	我感觉就是在不断'拉抽屉'
+    ID0115W0198	我公司员工不存在持有'和泰创投'股份的情况
+
+    ---->>
+    IC0010W0228 Here's
+    IC0012W0161 I'm
+    IC0013W0018 It's
+    IC0017W0126 Nothing'sGChange
+    IC0020W0392 She's
+    IC0022W0444 That's
+    IC0073W0058 搬不走的要及时关停并转
+    IC0085W0187 帮我放一首歌Let's
+    IC0392W0410 对低收入群体的帮助也更大
+    IC0975W0451 明年二月底小成
+    ID0114W0368 我感觉就是在不断拉抽屉
+    ID0115W0198 我公司员工不存在持有和泰创投股份的情况
+
+    """
+    new_line = []
+    line = list(line)
+    for i, char in enumerate(line):
+        if char == "'" and "\u4e00" <= line[i - 1] <= "\u9fff":
+            char = char.replace("'", "")
+        new_line.append(char)
+    line = "".join(new_line)
```

Comment:
Does it remove spaces between English words?

Reply:
@csukuangfj, No, it doesn't remove spaces between English Words.
I select some lines from trans.txt of the aishell2 dataset.
```
IC0001W0061     听流年
IC0001W0062     听beat it
IC0001W0063     听独角戏
IC0001W0064     听心雨
IC0001W0065     听Yesterday Once More
IC0001W0066     听广岛之恋
IC0001W0067     听一生有你
IC0010W0228     Here's
IC0012W0161     I'm
IC0013W0018     It's
IC0017W0126     Nothing'sGChange
IC0020W0392     She's
IC0022W0444     That's
IC0073W0058     搬不走的要及时'关停并转'
IC0085W0187     帮我放一首歌Let's
IC0392W0410     对低收入群体的帮助也更大'
IC0975W0451     明年二月底'小成'
ID0114W0368     我感觉就是在不断'拉抽屉'
ID0115W0198     我公司员工不存在持有'和泰创投'股份的情况                                                                  
```
After the code, it is as follows:
```
IC0001W0061 听流年
IC0001W0062 听BEAT IT
IC0001W0063 听独角戏
IC0001W0064 听心雨
IC0001W0065 听YESTERDAY ONCE MORE
IC0001W0066 听广岛之恋
IC0001W0067 听一生有你
IC0010W0228 HERE'S
IC0012W0161 I'M
IC0013W0018 IT'S
IC0017W0126 NOTHING'SGCHANGE
IC0020W0392 SHE'S
IC0022W0444 THAT'S
IC0073W0058 搬不走的要及时关停并转
IC0085W0187 帮我放一首歌LET'S
IC0392W0410 对低收入群体的帮助也更大
IC0975W0451 明年二月底小成
ID0114W0368 我感觉就是在不断拉抽屉
ID0115W0198 我公司员工不存在持有和泰创投股份的情况
```
```

Reply:
Thanks, I see.
It splits words into characters, with spaces being kept.

Reply:
@csukuangfj , the test code   is as follows:
```
#!/usr/bin/env python3

import sys

def normalization(line: str):
    new_line = []
    line = list(line)
    #print(line)
    for i, char in enumerate(line):
        if char == "'" and "\u4e00" <= line[i - 1] <= "\u9fff":
            char = char.replace("'", "")
        new_line.append(char)
    #print(new_line)
    line = "".join(new_line)
    line = line.upper()
    return line


if __name__ == "__main__":
    file = sys.argv[1]
    with open(file, "r" ) as f:
        for line in f:
            line = line.strip().split()
            content = " ".join(line[1:])
            content = normalization(content)
            print(f"{line[0]} {content}")
```

Reply:
> Thanks, I see. It splits words into characters, with spaces being kept.

Yes.